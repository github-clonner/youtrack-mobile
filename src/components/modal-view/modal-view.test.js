import React from 'react';
import {Text} from 'react-native';

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import ModalView from './modal-view';

describe('<ModalView/>', () => {

  let wrapper;

  describe('Render', () => {
    beforeEach(() => {
      wrapper = doShallow();
    });

    it('should match a snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render component', () => {
      expect(findByTestId('modalView')).toHaveLength(1);
    });
  });


  function findByTestId(testId) {
    return wrapper && wrapper.find({testID: testId});
  }

  function doShallow(
    {
      visible,
      transparent,
      animationType,
      supportedOrientations,
      onRequestClose
    } = {}
  ) {
    return shallow(
      <ModalView
        visible={visible}
        transparent={transparent}
        animationType={animationType}
        supportedOrientations={supportedOrientations}
        onRequestClose={onRequestClose}

      ><Text>children</Text>
      </ModalView>
    );
  }
});
