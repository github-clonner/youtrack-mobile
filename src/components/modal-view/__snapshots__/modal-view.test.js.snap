// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalView/> Render should match a snapshot 1`] = `
<Component
  animationType="none"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  testID="modalView"
  visible={true}
>
  <withOrientation
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {},
        ]
      }
    >
      <Text>
        children
      </Text>
    </View>
  </withOrientation>
</Component>
`;
